scanner/indexer/indexer.go

```diff
@@ -312,3 +297,70 @@ func GetDigestFromReference(ref name.Reference, auth authn.Authenticator) (name.
 	}
 	return dRef, nil
 }
+
+// CreateLibIndex handles the creation of libIndex.
+func CreateLibIndex(ctx context.Context, indexerCfg config.IndexerConfig, store indexer.Store, locker *ctxlock.Locker) (*libindex.Libindex, error) {
+	c := http.DefaultClient
+	faRoot, err := os.MkdirTemp("", "scanner-fetcharena-*")
+	if err != nil {
+		return nil, fmt.Errorf("creating indexer root directory: %w", err)
+	}
+	defer utils.IgnoreError(func() error {
+		if err != nil {
+			return os.RemoveAll(faRoot)
+		}
+		return nil
+	})
+
+	opts := libindex.Options{
+		Store:                store,
+		Locker:               locker,
+		FetchArena:           libindex.NewRemoteFetchArena(c, faRoot),
+		ScanLockRetry:        libindex.DefaultScanLockRetry,
+		LayerScanConcurrency: libindex.DefaultLayerScanConcurrency,
+	}
+
+	setupScannerConfig(&opts, indexerCfg)
+
+	indexer, err := libindex.New(ctx, &opts, c)
+	if err != nil {
+		return nil, fmt.Errorf("creating libindex: %w", err)
+	}
+
+	return indexer, nil
+}
+
+// setupScannerConfig sets up the scanner configuration.
+func setupScannerConfig(opts *libindex.Options, cfg config.IndexerConfig) {
+	rhelConfigs := map[string]RhelConfig{
+		"rhel-repository-scanner": {
+			ConfigKey: "repo2cpe",
+			// If CfgURL is empty, Claircore uses default URL: "https://access.redhat.com/security/data/metrics/repository-to-cpe.json",
+			CfgURL: cfg.RepositoryToCPEURL,
+		},
+		"rhel_containerscanner": {
+			ConfigKey: "name2repos",
+			// If CfgURL is empty, Claircore uses default URL: "https://access.redhat.com/security/data/metrics/container-name-repos-map.json",
+			CfgURL: cfg.NameToCPEURL,
+		},
+	}
+
+	opts.ScannerConfig.Repo = make(map[string]func(interface{}) error, len(rhelConfigs))
+	opts.ScannerConfig.Package = make(map[string]func(interface{}) error, len(rhelConfigs))
+
+	for scannerName, rcfg := range rhelConfigs {
+		jsonData := []byte(`{"` + rcfg.ConfigKey + `_mapping_url": "` + rcfg.CfgURL + `"}`)
+		switch scannerName {
+		case "rhel-repository-scanner":
```

Comment:
it seems like the purpose of `rhelConfigs` was to use it in some generic way, but it looks like we still need to special case each config anyway. I think we can simplify all this:

```
	opts.ScannerConfig.Repo = map[string]func(interface{}) error{
		"rhel-repository-scanner": makeScannerFunc(fmt.Sprintf("{repo2cpe_mapping_url: %s}", cfg.RepositoryToCPEURL)),
	}
	opts.ScannerConfig.Package = map[string]func(interface{}) error{
		"rhel_containerscanner": makeScannerFunc(fmt.Sprintf("{name2repos_mapping_url: %s}", cfg.RepositoryToCPEURL)),
	}
```

You probably don't even need the separation function for this, either. It's pretty short this way